srml/staking/src/lib.rs

```diff
@@ -552,54 +552,66 @@ decl_module! {
 
 			let mut ledger = Self::ledger(&controller).ok_or("not a controller")?;
 			let StakingLedger {
+				stash,
+				total_ring,
 				active_ring,
 				active_deposit_ring,
+				total_kton,
 				active_kton,
-				unlocking,
 				..
 			} = &mut ledger;
 
 			ensure!(
-				unlocking.len() < MAX_UNLOCKING_CHUNKS,
+				T::Ring::locks_count(stash) + T::Kton::locks_count(stash) < MAX_UNLOCKING_CHUNKS,
 				"can not schedule more unlock chunks"
 			);
 
-			let era = Self::current_era() + T::BondingDuration::get();
-
+			let at = <timestamp::Module<T>>::now().saturated_into::<TimeStamp>() + T::BondingDuration::get();
 			match value {
 				StakingBalance::Ring(r) => {
 					// total_active_ring = normal_ring + time_deposit_ring
 					// Only active normal ring can be unbond
 					let active_normal_ring = *active_ring - *active_deposit_ring;
 					// unbond normal ring first
-					let available_unbund_ring = r.min(active_normal_ring);
-
-					<RingPool<T>>::mutate(|r| *r -= available_unbund_ring);
-
-					if !available_unbund_ring.is_zero() {
-						*active_ring -= available_unbund_ring;
-						unlocking.push(UnlockChunk {
-							value: StakingBalance::Ring(available_unbund_ring),
-							era,
-						});
-
-						Self::update_ledger(&controller, &ledger, value);
+					let available_unbond_ring = r.min(active_normal_ring);
+
+					<RingPool<T>>::mutate(|r| *r -= available_unbond_ring);
+
+					if !available_unbond_ring.is_zero() {
+						*active_ring -= available_unbond_ring;
+						let expired_locks_ring = T::Ring::update_lock(
+							stash,
+							Some(CompositeLock::Unbonding(Lock {
+								amount: available_unbond_ring,
+								at,
+								reasons: WithdrawReasons::all()
+							})),
+						);
+						// TODO: check underflow?
+						*total_ring -= expired_locks_ring;
```

Comment:
Check: unbounding is not included in total_ring.

Reply:
These was the expired locks. In a 'Finish Unbond' state.